tests/test_parent_child_independent.py

```diff
@@ -0,0 +1,46 @@
+from tap_tester import runner, connections
+from base import SnapchatBase
+
+class SnapchatParentChildIndependentTest(SnapchatBase):
+
+    def name(self):
+        return "tap_tester_snapchat_ads_parent_child_test"
+
+    def test_run(self):
+        # select first_level_child_streams only and run test
+        first_level_child_streams = {"funding_sources", "billing_centers", "members", "roles", "ad_accounts", "product_catalogs"}
+        self.run_test(first_level_child_streams)
+
+        # select second_level_child_streams only and run test
+        second_level_child_streams = {"ad_account_stats_daily", "ad_account_stats_hourly", "audience_segments", "pixels", "media",
+                               "creatives", "phone_numbers", "campaigns", "ad_squads", "ads", "product_sets"}
+        # Don't have data for stats stream so excluding from test
+        self.run_test(second_level_child_streams - self.stats_streams)
+        
+        # select third_level_child_streams only and run test
+        third_level_child_streams = {"pixel_domain_stats", "campaign_stats_daily", "campaign_stats_hourly", "ad_squad_stats_daily",
+                                     "ad_squad_stats_hourly", "ad_stats_daily", "ad_squad_stats_hourly"}
+        # Don't have data for stats streams so excluding from test(all streams are stats stream for third_level_child_streams)
+        # self.run_test(third_level_child_streams - self.stats_streams)    
```

Comment:
Could you adjust this pattern to create a new `test_` method for each case that then calls the `run_test`.

Reply:
Updated pattern as suggested by adding test_ method for all three scenarios.